From e35176a69e74e5eb7f80af8c68d1862b279d0a20 Mon Sep 17 00:00:00 2001
From: Markus Raab <elektra@markus-raab.org>
Date: Sun, 24 May 2015 20:55:35 +0200
Subject: [PATCH 5/7] fix build without full and static

plugin tests need to be disabled then,
most tests directly use functions from plugin
always use LIBRT_LIBRARIES
---
 cmake/Modules/LibAddMacros.cmake | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/cmake/Modules/LibAddMacros.cmake b/cmake/Modules/LibAddMacros.cmake
index ce7f15c..e8fe53c 100644
--- a/cmake/Modules/LibAddMacros.cmake
+++ b/cmake/Modules/LibAddMacros.cmake
@@ -106,7 +106,7 @@ endmacro()
 # links the executeable (only if build_full)
 # and adds a test
 macro (add_plugintest testname)
-	if (BUILD_TESTING)
+	if (BUILD_TESTING AND (BUILD_STATIC OR BUILD_FULL))
 		parse_arguments(ARG
 			"" # no arguments
 			"MEMLEAK" #options
@@ -135,7 +135,7 @@ macro (add_plugintest testname)
 			set_property(TEST testmod_${testname} PROPERTY
 				LABELS memleak)
 		endif (ARG_MEMLEAK)
-	endif (BUILD_TESTING)
+	endif ()
 endmacro (add_plugintest)
 
 # Add a test for cpp plugins
@@ -173,7 +173,7 @@ macro (add_cpp_plugintest testname)
 			set_property(TEST testmod_${testname} PROPERTY
 				LABELS memleak)
 		endif (ARG_MEMLEAK)
-	endif(BUILD_TESTING)
+	endif()
 endmacro (add_cpp_plugintest testname)
 
 
-- 
2.3.6

